addons/website_event_sale/controllers/main.py

```diff
@@ -70,3 +71,16 @@ def registration_confirm(self, event, **post):
                 request.website.sale_reset()
 
         return res
+
+
+class PaymentPortalOnsite(PaymentPortal):
+
+    def _validate_transaction_for_order(self, transaction, sale_order_id):
+        """
+        Throws a ValidationError if the user tries to pay for a ticket which isn't available
+        """
+        super()._validate_transaction_for_order(transaction, sale_order_id)
+        sale_order = request.env['sale.order'].browse(sale_order_id).exists().sudo()
+        events = sale_order.order_line.filtered(lambda line: line.product_type == 'event').mapped('event_ticket_id')
+        for event in events:
+            event._check_seats_availability(minimal_availability=sale_order.attendee_count)
```

Comment:
Welp, they're not events anymore ^^

Also you can just call with a recordset, the method supports recordsets (it's potentially better optimized this way)